bin/plugin/commands/build-plugins.js

```diff
@@ -0,0 +1,183 @@
+const { getModuleDataFromHeader, getModuleHeader } = require( './common' );
+
+exports.options = [];
+
+/**
+ * Command for copying the files required for the standalone plugin.
+ */
+exports.handler = async () => {
+	const pluginsFile = path.join( '.', 'plugins.json' );
+	fs.readFile( pluginsFile, 'utf8', ( err, jsonString ) => {
+		if ( err ) {
+			log(
+				formats.error(
+					`Error reading file from disk: "${ err }"`
+				)
+			);
+		}
+
+		try {
+			const plugins = JSON.parse( jsonString );
+			for ( const moduleDir in plugins ) {
+				const pluginVersion = plugins[ moduleDir ]?.version;
+				const pluginSlug = plugins[ moduleDir ]?.slug;
+
+				if ( ! pluginVersion || ! pluginSlug ) {
+					log(
+						formats.error(
+							'The given module configuration is invalid, the module JSON object is missing the "version" and/or "slug" properties, or they are misspelled.'
+						)
+					);
+					return;
+				}
+
+				try {
+					// Copy module files from the folder.
+					const modulePath = path.join( '.', 'modules/' + moduleDir );
+					const buildModulePath = path.join( '.', 'build/' + pluginSlug );
+					try {
+						// Clean up build module files directory.
+						fs.rmSync( buildModulePath, { force: true, recursive: true } );
+
+						fs.copySync( modulePath, buildModulePath, { overwrite: true } );
+					} catch ( copyError ) {
+						log(
+							formats.error(
+								`Error copying files for plugin "${ pluginSlug }": "${ copyError }"`
+							)
+						);
+						continue;
+					}
+
+					// Update file content.
+					updatePluginHeader( {
+						modulePath: moduleDir,
+						slug: pluginSlug,
+						version: pluginVersion,
+						pluginPath: buildModulePath,
+					} );
+
+					// Update text domain.
+					updateModuleDetails( {
+						pluginPath: buildModulePath,
+						regex: '[\']performance-lab[\']',
+						result: `'${ pluginSlug }'`,
+					} );
+
+					// Update `@package`.
+					updateModuleDetails( {
+						pluginPath: buildModulePath,
+						regex: '@package\\s{1,}performance-lab',
+						result: '@package ' + pluginSlug,
+					} );
+				} catch ( error ) {
+					log(
+						formats.error(
+							`${ error }`
+						)
+					);
+				}
+			}
+		} catch ( jsonError ) {
+			log(
+				formats.error(
+					`Error parsing JSON string: "${ jsonError }"`
+				)
+			);
+		}
+	} );
+};
+
+/**
+ * Updates the `load.php` file header information.
+ *
+ * @param {Object} settings Plugin settings.
+ */
+async function updatePluginHeader( settings ) {
+	const { modulePath, version, slug, pluginPath } = settings;
+	// Specific module `load.php` file content.
+	const buildLoadFile = path.join( pluginPath, 'load.php' );
+	let buildLoadFileContent = '';
+	try {
+		buildLoadFileContent = fs.readFileSync( buildLoadFile, 'utf-8' );
+	} catch ( err ) {
+		log(
+			formats.error(
+				`Error reading the file "${ buildLoadFile }": "${ err }"`
+			)
+		);
+	}
+
+	const moduleHeader = await getModuleHeader( buildLoadFileContent );
```

Comment:
@mukeshpanchal27 if we are setting `buildLoadFileContent` to an empty string by default, further above, we probably want to check for an empty string before proceeding?

Thinking some more around this, if there is a problem and the file can not be read, and therefore defaults to `''`, how does this affect the rest of the codebase where this `updatePluginHeader` function is called? Do we need to return `false` or similar and have conditional checks in areas that call this so that logic that follows calls to this function does not apply?

Keen to hear your thoughts.

Reply:
+1, we should do an early `return` after logging the error since the remaining logic otherwise won't work anyway.

Reply:
Thanks both for the feedback. Addressed in https://github.com/WordPress/performance/pull/662/commits/2b0507034912df7a457fc89f6158a473c88d5a4f